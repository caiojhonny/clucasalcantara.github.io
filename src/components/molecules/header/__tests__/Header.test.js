import React from 'react'
import { shallow } from 'enzyme'
import Header from '..'

describe('Header component', () => {
  it('should renders without crash', () => {
    const wrapper = shallow(
      <Header
        navItems={[{ text: 'Work with coffee', path: '/' }]}
        title="Title"
        subtitle="Subtitle"
        theme={{ colors: {} }}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
