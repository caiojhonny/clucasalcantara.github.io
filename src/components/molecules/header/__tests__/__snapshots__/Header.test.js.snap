// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    navItems={
      Array [
        Object {
          "path": "/",
          "text": "Work with coffee",
        },
      ]
    }
    subtitle="Subtitle"
    theme={
      Object {
        "colors": Object {},
      }
    }
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <_default
          content="Title"
          subtitle="Subtitle"
        />,
        <_default
          navItems={
            Array [
              Object {
                "path": "/",
                "text": "Work with coffee",
              },
            ]
          }
        />,
      ],
      "darkMode": undefined,
      "navItems": Array [],
      "palette": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": undefined,
          "content": "Title",
          "subtitle": "Subtitle",
          "subtitleColor": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isMobile": undefined,
          "navItems": Array [
            Object {
              "path": "/",
              "text": "Work with coffee",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": "1rem",
          "position": "sticky",
          "textTransform": "uppercase",
        },
      ],
      "defaultProps": Object {
        "navItems": Array [],
      },
      "displayName": "Styled(div)",
      "propTypes": Object {
        "isMobile": [Function],
        "navItems": [Function],
        "subtitle": [Function],
        "theme": [Function],
        "title": [Function],
      },
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <_default
            content="Title"
            subtitle="Subtitle"
          />,
          <_default
            navItems={
              Array [
                Object {
                  "path": "/",
                  "text": "Work with coffee",
                },
              ]
            }
          />,
        ],
        "darkMode": undefined,
        "navItems": Array [],
        "palette": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": undefined,
            "content": "Title",
            "subtitle": "Subtitle",
            "subtitleColor": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isMobile": undefined,
            "navItems": Array [
              Object {
                "path": "/",
                "text": "Work with coffee",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginBottom": "1rem",
            "position": "sticky",
            "textTransform": "uppercase",
          },
        ],
        "defaultProps": Object {
          "navItems": Array [],
        },
        "displayName": "Styled(div)",
        "propTypes": Object {
          "isMobile": [Function],
          "navItems": [Function],
          "subtitle": [Function],
          "theme": [Function],
          "title": [Function],
        },
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
